components/formats-gpl/src/loci/formats/in/CellSensReader.java

```diff
@@ -771,8 +771,29 @@ else if (zCount > 0) {
       if (s < files.size() - 1) {
         setCoreIndex(index);
         String ff = files.get(s);
+        /**
+         * If there are more frame_*.ets files than there are metadata 'pyramids'
+         * defined in the .vsi, then we have orphaned frame files.  In this case
+         * we need to determine which are the valid frame files and which are 
+         * the orphans.  The valid ones need to be matched with the appropriate
+         * metadata blocks in the vsi, and the orphaned ones need to be ignored.
+        */
+        boolean hasOrphanEtsFiles = pyramids.size() < (files.size() - 1);
         try (RandomAccessInputStream stream = new RandomAccessInputStream(ff)) {
-            parseETSFile(stream, ff, s);
+            boolean validFrameFile = parseETSFile(stream, ff, s, hasOrphanEtsFiles);
+            /*
+             * If this frame file is an orphan, "undo" the work that was done for
+             * it and change its status to being an "extra" file.
+             */
+             if (!validFrameFile) {
+                core.remove(core.size()-1);
```

Comment:
```suggestion
                fileMap.remove(core.size() - 1);
                core.remove(core.size()-1);
```

Reply:
Good catch Melissa!  I tried to be thorough but that one escaped me.  I place the filemap.remove statement within the parseETSFile() function; it seemed more appropriate since the file was added to the filemap from within that function.  Please let me know if you encounter any other issues.  Thank you.